### PR TITLE
legacyrpc: add fundrawtransaction command

### LINKED ISSUE #1181: Implement fundrawtransaction JSON-RPC
This method is currently unimplemented, and should be implemented as part of the raw transaction JSON-RPC API.

The behavior should be similar to [Bitcoin Core's implementation](https://github.com/bitcoin/bitcoin/blob/ca2a23387be6e4b18a68721306c47dc4b5582a93/src/wallet/rpcwallet.cpp#L3371-L3425).  Some differences that must be accounted for are:

* The funding account must be specified as a non-optional parameter
* Address types (for change) can likely be ignored, since we have a malleability fix for normal P2PKH
* We don't have separate watching-only addresses in wallets with other private keys, so `includeWatching` makes little sense
* RBF (BIP125) isn't implemented by dcrwallet and doesn't need support in this request
* There is no fee rate estimation, so the `estimate_mode` parameter doesn't make sense
* There is no segwit distinction in Decred, so the `iswitness` parameter is unnecessary

### PR DESCRIPTION
**Pull request by vctt94**
closes #1181

### CODE REVIEW

File: rpc/legacyrpc/methods.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ import (
 	"github.com/decred/dcrwallet/wallet"
 	"github.com/decred/dcrwallet/wallet/txrules"
 	"github.com/decred/dcrwallet/wallet/udb"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
```
Comment (jrick) [inline rpc/legacyrpc/methods.go]: there should be no grpc imports in the json-rpc implementation (and likewise, no json-rpc code in the grpc server)

Comment (vctt94) [reply rpc/legacyrpc/methods.go]: fixed

**Comment by vctt94** [comment]
This PR is ready for another review

**Comment by vctt94** [comment]
Thanks for reviewing my PR again Jrick. 

I fixed your comments and this PR is ready for another review.

File: wallet/txauthor/author.go [hunk 2]
```diff
@@ -154,6 +154,82 @@ func NewUnsignedTransaction(outputs []*wire.TxOut, relayFeePerKb dcrutil.Amount,
...
 }
 
+func FundRawTransaction(inputs []*wire.TxIn, outputs []*wire.TxOut, relayFeePerKb dcrutil.Amount,
+	fetchInputs InputSource, fetchChange ChangeSource) (*AuthoredTx, error) {
+
+	const op errors.Op = "txauthor.FundRawTransaction"
+
+	totalInTransaction := h.SumInputValues(inputs)
+	targetAmount := h.SumOutputValues(outputs) - totalInTransaction
+	scriptSizes := []int{txsizes.RedeemP2PKHSigScriptSize}
```
[the remark this answers is not in the document]
Comment (vctt94) [reply wallet/txauthor/author.go]: I changed the initial scriptSizes to this plus each input the transaction might be, is that right now?

```
scriptSizes := []int{txsizes.RedeemP2PKHSigScriptSize}
for range mtx.TxIn {
	scriptSizes = append(scriptSizes, txsizes.RedeemP2PKHInputSize)
}
```

File: wallet/txauthor/author.go [hunk 3]
```diff
@@ -154,6 +154,75 @@ func NewUnsignedTransaction(outputs []*wire.TxOut, relayFeePerKb dcrutil.Amount,
...
 }
 
+func FundRawTransaction(mtx wire.MsgTx, relayFeePerKb dcrutil.Amount,
+	fetchInputs InputSource, fetchChange ChangeSource) (*wire.MsgTx, error) {
+
+	const op errors.Op = "txauthor.FundRawTransaction"
+
+	inputs := mtx.TxIn
+	outputs := mtx.TxOut
+	totalInTransaction := h.SumInputValues(inputs)
```
[the remark this answers is not in the document]
Comment (vctt94) [reply wallet/txauthor/author.go]: Not sure what this `wire.NullValuIn` is supposed to do. I added a verification checking if the `TxIn` array is not empty, is that right? `if len(inputs) > 0 {`

**Comment by vctt94** [comment]
I added the fee estimation as we discussed and this PR is ready for another review.

**Comment by vctt94** [comment]
rebased and moved `fundrawtransaction` to `jsonrpc/types` modules

**Comment by jrick** [comment]
Implemented in #1706

**Pull request closed without merging**